weedcoco/repo/deposit.py

```diff
@@ -129,17 +130,28 @@ def compress_to_download(dataset_dir, deposit_id, download_dir):
         copy(zip_path, download_path)
 
 
+def atomic_copy(repository_dir, download_dir, temp_dir, deposit_id):
+    dataset_path = temp_dir / deposit_id
+    zip_path = temp_dir / f"{deposit_id}.zip"
+    repository_dir = repository_dir / deposit_id
+    if os.path.isdir(dataset_path):
+        move(str(dataset_path), str(repository_dir))
+    if os.path.isfile(zip_path):
+        move(str(zip_path), str(download_dir))
```

Comment:
To make this a little safer:
```suggestion
    try:
        if os.path.isfile(zip_path):
            move(str(zip_path), str(download_dir))
        if os.path.isdir(dataset_path):
            move(str(dataset_path), str(repository_dir))
    except:
        # roll back to leave a clean repository
        (download_dir / zip_path).unlink(missing_ok=True)
        for path in (repository_dir / deposit_id).glob("*"):
            path.unlink(missing_ok=True)
        (repository_dir / deposit_id).rmdir()
        raise
```